source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,266 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding a subset of them in memory at a given
+time. The Go driver uses a cursor in some read operations to return
+matched documents in batches as opposed to all at once.
+
+Retrieve Sample Data
+~~~~~~~~~~~~~~~~~~~~
+
+Each section assumes you retrieve all the documents in your collection
+with the ``Find()`` function.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+Access Cursor Elements
+----------------------
+
+You can access a cursors elements before it's **exhausted**. A cursor is
+exhausted when the ``MongoClient`` iterates through its results and
+reaches its last element. Afterwards, the cursor won't respond to
+functions to access its elements.
+
+.. important::
+
+   A cursor is not goroutine safe. Do not use the same cursor in
+   multiple goroutines at the same time.
+
+Current Document
+~~~~~~~~~~~~~~~~
+
+When :ref:`iterating through your cursors results
+<cursor-iteration-golang>`, you can view each document as a ``BSON`` or
+Go value.
+
+BSON
+````
```

Comment:
Issue: [Three levels of headings](https://docs.mongodb.com/meta/style-guide/style/titles-and-headings/#style-and-structure)